src/streamlink/plugins/ard_mediathek.py

```diff
@@ -35,54 +13,63 @@
     r"https?://(?:(\w+\.)?ardmediathek\.de/|mediathek\.daserste\.de/)"
 ))
 class ARDMediathek(Plugin):
-    def _get_http_streams(self, info):
-        name = QUALITY_MAP.get(info["_quality"], "vod")
-        urls = info["_stream"]
-        if not isinstance(info["_stream"], list):
-            urls = [urls]
+    def _get_streams(self):
+        data_json = self.session.http.get(self.url, schema=validate.Schema(
+            validate.parse_html(),
+            validate.xml_findtext(".//script[@id='fetchedContextValue'][@type='application/json']"),
+            validate.any(None, validate.all(
+                validate.parse_json(),
+                {str: dict},
+                validate.transform(lambda obj: list(obj.items())),
+                validate.filter(lambda item: item[0].startswith("https://api.ardmediathek.de/page-gateway/pages/")),
+                validate.any(validate.get((0, 1)), [])
+            ))
+        ))
+        if not data_json:
+            return
 
-        for url in urls:
-            stream = HTTPStream(self.session, update_scheme("https://", url))
-            yield name, stream
+        schema_data = validate.Schema({
+            "id": str,
+            "widgets": validate.all(
```

Comment:
This breaks on age restricted content before 22:00 in Germany as `widgets` isn't returned then.

Content like [this](https://www.ardmediathek.de/video/fernsehfilm/corpus-christi-oder-drama/wdr/Y3JpZDovL3dkci5kZS9CZWl0cmFnLTVjMTEwZjJmLTUzOTMtNDBlYS04OTRlLWUwZWJkOGM2OGVlOQ/).

Reply:
Or maybe there is no `mediaCollection`, I can't remember and sadly I've lost the output of the test with your plugin.  Either way, it broke earlier, so it might be worth checking out tomorrow before 22:00.  I don't know what time in the morning the age restriction kicks back in.

Sorry about the vagueness of this info.  I wish I still had the output from the run earlier to post here.

Reply:
No, thanks for letting me know. That's why I would've appreciated a review first instead of an instant merge.
There's some content which expires in less than an hour. I'll have a look then.

Reply:
I'll have to check tomorrow. I could open a PR with my changes now, but I want to be sure first. I've also applied stuff from my ARDLive plugin rewrite a couple of weeks ago which I should've checked immediately.